src/http-client/node-http2-client.ts

```diff
@@ -4,47 +4,106 @@ try {
 
-  /**
-   * @remarks Private constructor means you must instantiate with
-   * {@link NodeHTTP2Client.getClient}
-   */
-  // eslint-disable-next-line @typescript-eslint/no-empty-function
-  private constructor() {}
+  #http2_session_idle_ms: number;
+  #url: string;
 
-  static getClient() {
+  constructor({ http2_session_idle_ms, url }: HTTPClientOptions) {
     if (http2 === undefined) {
       throw new Error("Your platform does not support Node's http2 library");
     }
 
-    if (this.#client === null) {
-      this.#client = new NodeHTTP2Client();
-    }
-
-    this.#client.#numberOfUsers++;
-    return this.#client;
+    this.#http2_session_idle_ms = http2_session_idle_ms;
+    this.#url = url;
   }
 
   /** {@inheritDoc HTTPClient.request} */
-  async request(httpRequest: HTTPRequest): Promise<HTTPResponse> {
-    const session = this.#getSession(httpRequest.url);
+  async request({
+    client_timeout_ms,
+    data: requestData,
+    headers: requestHeaders,
+    method,
+  }: HTTPRequest): Promise<HTTPResponse> {
+    let req: ClientHttp2Stream;
+
+    const requestPromise = new Promise<HTTPResponse>(
+      (resolvePromise, rejectPromise) => {
+        const onResponse = (
+          http2ResponseHeaders: IncomingHttpHeaders & IncomingHttpStatusHeader
+        ) => {
+          const status = Number(
+            http2ResponseHeaders[http2.constants.HTTP2_HEADER_STATUS]
+          );
+          let responseData = "";
+
+          // append response data to the data string every time we receive new
+          // data chunks in the response
+          req.on("data", (chunk: any) => {
+            responseData += chunk;
+          });
+
+          // Once the response is finished, resolve the promise
+          req.on("end", () => {
+            resolvePromise({
+              status,
+              body: responseData,
+              headers: http2ResponseHeaders,
+            });
+          });
+        };
+
+        try {
+          const httpRequestHeaders: OutgoingHttpHeaders = {
+            ...requestHeaders,
+            [http2.constants.HTTP2_HEADER_PATH]: "/query/1",
+            [http2.constants.HTTP2_HEADER_METHOD]: method,
+          };
+
+          const session = this.#connect();
+          req = session
+            .request(httpRequestHeaders)
+            .setEncoding("utf8")
+            .on("error", (error: any) => {
+              rejectPromise(error);
+            })
+            .on("response", onResponse);
+
+          req.write(JSON.stringify(requestData), "utf8");
+
+          // req.setTimeout must be called before req.end()
+          req.setTimeout(client_timeout_ms, () => {
+            req.destroy(new Error(`Client timeout`));
+          });
+
+          req.end();
+        } catch (error) {
+          rejectPromise(error);
+        }
+      }
+    );
 
     try {
-      const result = await session.request(httpRequest);
-      return result;
+      return await requestPromise;
```

Comment:
no need to return await.

```suggestion
      return requestPromise;
```